robottelo/containers.py

```diff
@@ -0,0 +1,249 @@
+            ports=None
+    ):
+        if runtime == "docker":
+            ContainerClass = DockerContainer
+        # elif runtime == "podman":
+        #     ContainerClass = PodmanContainer
+        else:
+            logger.warning("Invalid runtime selection")
+        self._inst = ContainerClass(image=image, tag=tag, agent=agent,
+                                    ports=ports)
+        self._logs = self._inst._logs
+
+    @property
+    def name(self):
+        return self._inst.name
+
+    def delete(self):
+        return self._inst.delete()
+
+    def execute(self, command):
+        return self._inst.execute(command)
+
+    def logs(self, file="stdout", tailing=False):
+        return self._inst.logs(file, tailing)
+
+    def port(self):
+        return self._inst.port()
+
+    def register(
+            self,
+            host=settings.server.hostname,
+            ak=None,
+            org="Default_Organization",
+            env="Library",
+            auth=("admin", "changeme"),
+            force=False,
+    ):
+        self._host = host
+        res = self._inst.execute(
+            f"curl --insecure --output katello-ca-consumer-latest.noarch.rpm \
+                https://{host}/pub/katello-ca-consumer-latest.noarch.rpm"
+        )
+        res += self._inst.execute(
+            "yum -y localinstall katello-ca-consumer-latest.noarch.rpm"
+        )
+        if "Complete!" not in res:
+            raise ContainerError(f'Unable to install bootstrap rpm\n{res}')
+        reg_command = f'subscription-manager register --org="{org}"'
+        if force:
+            reg_command += " --force"
+        if ak:
+            res += self._inst.execute(f'{reg_command} --activationkey="{ak}"')
+        else:
+            res += self._inst.execute(
+                f'{reg_command} --user="{auth[0]}" --password="{auth[1]}" \
+                --environment="{env}"'
+            )
+        if "The system has been registered" not in res:
+            raise ContainerError('Unable to register to system' +
+                                 '/n output from container:' + res)
+        return res
+
+    def rex_setup(self, host=None):
+        self._inst.execute("mkdir /root/.ssh")
+        return self._inst.execute(
+            f"curl -ko /root/.ssh/authorized_keys \
+                https://{host or self._host}:9090/ssh/pubkey"
+        )
+
+    def install_katello_agent(self):
+        """Installs katello agent on the Container.
+        :return: result of stdout of installation katello-agent and goferd
+        process check.
+        :raises ContainerError: If katello-ca wasn't installed.
+        """
+        result = self._inst.execute("yum install -y katello-agent")
+
+        result += self._inst.execute('rpm -q katello-agent')
+        if "Complete!" not in result:
+            raise ContainerError('Failed to install katello-agent')
+        if not self._inst._mount:
+            raise ContainerError('katello-agent is not running')
```

Comment:
how are you coming on the conclusion that katello-agent is not running? I don't see anywhere checking of gofer process status